pytest_mh/utils/tc.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import annotations
+
+from .. import MultihostHost, MultihostUtility
+from ..ssh import SSHLog
+
+__all__ = ["LinuxTrafficControl"]
+
+
+class LinuxTrafficControl(MultihostUtility):
+    """
+    Perform traffic control operations on remote host.
+
+    All changes are automatically reverted when a test is finished.
+    """
+
+    def __init__(self, host: MultihostHost) -> None:
+        """
+        :param host: Remote host instance.
+        :type host: MultihostHost
+        """
+        super().__init__(host)
+        self.__interfaces: set[str] = set()
+        self.__restore: set[str] = set()
+
+    def setup_when_used(self):
+        """
+        Setup traffic control configuration.
+
+        :meta private:
+        """
+        super().setup_when_used()
+
+        # Let's find out the availabel interfaces
+        result = self.host.ssh.run("ip -o address", log_level=SSHLog.Error)
+        split_result = result.stdout.splitlines()
+
+        for line in split_result:
+            interface = line.split(" ")[1]
+            if interface != "lo":
+                self.__interfaces.add(interface)
+
+    def teardown_when_used(self):
+        """
+        Revert all traffic control changes.
+
+        :meta private:
+        """
+        tear: str = ""
+        for iter in self.__restore:
+            tear += iter + "\n"
+        self.host.ssh.run(tear, log_level=SSHLog.Error)
+        super().teardown_when_used()
+
+    def add_delay(self, hostname: str, time: int):
```

Comment:
One question for the last case, should I use the same band for all ip addresses in the list?

Reply:
Since the delay is the same for all addresses, then yes.

But actually, you this was a leftover comment from some previous review session. I think this change would make it more generic, but I don't think it is really needed and I am happy with latest interface. Feel free to disregard it.